presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -215,10 +213,10 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
             return WriteMapping.longMapping("number(19,0)", bigintWriteFunction());
         }
         if (type instanceof TimestampType) {
-            return WriteMapping.longMapping(format("timestamp(%s)", timestampDefaultPrecision), timestampWriteFunction(session));
+            return WriteMapping.longMapping("timestamp(3)", timestampWriteFunction(session));
```

Comment:
why 3.. oracle default is 6.
we shouldn't carry a limitation of presto timestamp to oracle

Reply:
`timestamp(6)` will lead to incorrect results (different results when predicate is pushed down and when it is not)

Anyway, this will be addressed soon the right way with the ongoing work at https://github.com/prestosql/presto/issues/1284

Reply:
so any table create outside presto will have that problem (it will be created as timestamp(6))

Reply:
correct